### PR TITLE
Cache refactor

### PR DESCRIPTION
**Pull request by 2qar**
Messy PR that does a couple things

1. Refactors the access token + Trackmania cache code to use the same cursed Java code
2. Created missing Trackmania OAuth service class. Now it has token caching + fetches through the same BaseService.fetchEndpoint as the other service API classes
3. Puts all of the Trackmania services (Core, Live, Meet, OAuth) behind a facade. There's more cleanup that could be done so you never need to access the client for a service directly, but do wat you wanna do :sunglasses: 

:shrimp:

### CODE REVIEW

**Comment by Khujou** [comment]
leaderboard button doesnt work! fix it NOW

File: app.js [hunk 1]
```diff
@@ -232,7 +222,8 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
 
             await DiscordRequest(endpoint, {
                 method: 'PATCH',
-                body: await trackmania.leaderboard(live_service, track_info, args[3], true, args[4]),
+                body: await trackmania.leaderboard(trackmaniaFacade.liveService,
+                    trackmaniaFacade.oauthService, track_info, args[3], true, args[4]),
```
Comment (Khujou) [inline app.js:226]: you seem to be giving this function more arguments than it can take

**Comment by 2qar** [comment]
> leaderboard button doesnt work! fix it NOW

feexed

File: trackmania.js [hunk 2]
```diff
@@ -119,22 +183,61 @@ class BaseService {
      * @returns {Promise<JSON>}
      */
     async fetchEndpoint(endpoint) {
-        return await fetch(this.url + endpoint, {
-            headers: {
-                "User-Agent": 'trackmania-bot Discord Bot : https://github.com/Khujou/trackmania-bot',
-                Authorization: `nadeo_v1 t=${await this.getAccessToken().then(response => response.accessToken)}`,
-            }
+        console.log(`Fetching endpoint "${this.baseUrl + endpoint}"`);
+        const finalEndpoint = this.baseUrl + endpoint;
```
[the remark this answers is not in the document]
Comment (2qar) [reply trackmania.js:195]: You are so right. I will poosh this

**Pull request merged**